### DISCUSSION TITLE
broker的slave节点消费太慢问题

### DISCUSSION THREAD

**Discussion by 18191518054** [apache/rocketmq#7847] [Show and tell]
我从文档中了解到Broker的Slave节点负责备份 Master 的数据，以及备用选举。那为什么消费者可以监听brokerId是非0的节点(Slave),当我在节点slave节点时，消费者获取消息会非常慢，通过Arthas定位分析出具体原因，代码如下：
![image](https://github.com/apache/rocketmq/assets/69226167/d0836802-2a55-446d-b32c-f9e84f0c763e)
其中就有一个判断，当节点非SLAVE是，才会通知新消息到来，这样是不是可以说明我们只能监听主节点进行消费？

**Comment by 18191518054** [comment]
还有一个疑问，我使用的版本是4.9.4这里通知新消息到达会通知两次
![image](https://github.com/apache/rocketmq/assets/69226167/5e129384-cd14-4ba1-acd5-bd42d6786a37)
，看了一下4.5.0版本，此版本只会通知一次，没有notifyMessageArrive4MultiQueue()方法的通知，这是BUG修复了吗

**Reply by cserwen** [reply]
1. 请求到从节点属于从读功能，默认实现只有客户端读取比较老的数据时才会访问从节点，新数据还是会从 Master 拉取，因此这个消息通知机制没有问题，从读时必定存在堆积。
2. 4.9 版本的 notifyMessageArrive4MultiQueue() 是为了 LMQ 功能实现的，不是修复这个问题，况且仍旧是在 Master 状态运行的:-D @18191518054
